plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/BaseDeltaLakeConnectorSmokeTest.java

```diff
@@ -2161,6 +2167,55 @@ public void testPartitionFilterIncluded()
         }
     }
 
+    @Test(dataProviderClass = DataProviders.class, dataProvider = "trueFalse")
+    public void testConcurrentInsertsReconciliation(boolean partitioned)
+            throws Exception
+    {
+        int threads = 3;
+
+        CyclicBarrier barrier = new CyclicBarrier(threads);
+        ExecutorService executor = newFixedThreadPool(threads);
+        String tableName = "test_concurrent_inserts_table_" + randomNameSuffix();
+
+        assertUpdate("CREATE TABLE " + tableName + " (a INT, part INT) " +
+                (partitioned ? " WITH (partitioned_by = ARRAY['part'])" : ""));
+
+        try {
+            // insert data concurrently
+            executor.invokeAll(ImmutableList.<Callable<Void>>builder()
+                            .add(() -> {
+                                barrier.await(20, SECONDS);
+                                getQueryRunner().execute("INSERT INTO " + tableName + " VALUES (1, 10)");
+                                return null;
+                            })
+                            .add(() -> {
+                                barrier.await(20, SECONDS);
+                                getQueryRunner().execute("INSERT INTO " + tableName + " VALUES (11, 20)");
```

Comment:
Do we check type of changes made by another call somewhere? If we assume that any change to data (`UPDATE`, `MERGE`, `DELETE`) is fine it would be good to add such test scenarios.

Reply:
This would be a follow-up https://github.com/trinodb/trino/pull/18521